src/vs/workbench/browser/parts/statusbar/statusbarPart.ts

```diff
@@ -287,37 +286,20 @@ class StatusBarEntryItem implements IStatusbarItem {
 	private executeCommand(id: string, args?: any[]) {
 		args = args || [];
 
-		// Lookup built in commands
-		const builtInActionDescriptor = Registry.as<IWorkbenchActionRegistry>(ActionExtensions.WorkbenchActions).getWorkbenchAction(id);
-		if (builtInActionDescriptor) {
-			const action = this.instantiationService.createInstance(builtInActionDescriptor.syncDescriptor);
-
-			if (action.enabled) {
-				/* __GDPR__
-					"workbenchActionExecuted" : {
-						"id" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
-						"from": { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
-					}
-				*/
-				this.telemetryService.publicLog('workbenchActionExecuted', { id: action.id, from: 'status bar' });
-				(action.run() || TPromise.as(null)).done(() => {
-					action.dispose();
-				}, (err) => this.messageService.show(Severity.Error, toErrorMessage(err)));
-			} else {
-				this.messageService.show(Severity.Warning, nls.localize('canNotRun', "Command '{0}' is currently not enabled and can not be run.", action.label || id));
-			}
-
-			return;
-		}
-
 		// Maintain old behaviour of always focusing the editor here
 		const activeEditor = this.editorService.getActiveEditor();
 		const codeEditor = getCodeEditor(activeEditor);
 		if (codeEditor) {
 			codeEditor.focus();
```

Comment:
Not sure why we have it... Maybe because the command palette does that and because enablement of commands often depends on editor focus...

Reply:
@jrieken I think this would not be needed if all commands have a proper when clause and would simply not show up when not enabled.

Reply:
Well, the problem is that selecting something from the status bar takes/steels focus and that this means you cannot have things like 'comment line' in there. Not sure if they should find and focus an editor when executed...